app/Library/Notifications/RecipientFinder.php

```diff
@@ -59,6 +61,21 @@ public function publishedNewTrack(Track $track) {
         switch ($this->notificationDriver) {
             case PonyfmDriver::class:
                 return $track->user->followers;
+            case NativeDriver::class:
+                $followerIds = [];
+                $subIds = [];
+                $rawSubIds = Subscription::select('id')->get();
+
+                foreach ($track->user->followers as $follower) {
+                    array_push($followerIds, $follower->id);
+                }
+
+                foreach ($rawSubIds as $sub) {
+                    array_push($subIds, $sub->id);
+                }
+
+                $targetIds = array_intersect($followerIds, $subIds);
```

Comment:
It should be possible to build this list of target ID's using an inner join database query.